### PR TITLE
Added place methods and Added overlap test

### PR DESCRIPTION
**Pull request by sasloan**
@lrs8810 I created two new test's one is that the ship can be placed and one is that the ship does not over lap then I wrote a place method. also this pull request has the final version of our valid_placement? method. and valid_placement test. If you have a moment please review my place method and the two test and if they look good then please approve and if not please let me know what questions you have.

### CODE REVIEW

File: lib/board.rb [hunk 1]
```diff
@@ -88,38 +74,23 @@ def valid_column?(coordinates)
...
-    return false unless coordinates.length == ship.length
+    return false unless ship.length == coordinates.length
     coordinates.each_cons(2).all? {|a, b| b[0].ord == a[0].ord + 1 }
   end
->>>>>>> fe4c28bec72789a09f42a9fcbfec700a43886c39
-=======
->>>>>>> ss_board
+
+  def place(ship, coordinates)
+    if valid_placement?(ship, coordinates)
```
Comment (lrs8810) [inline lib/board.rb]: This looks like a great method!

Comment (sasloan) [reply lib/board.rb]: awesome thanks!!!

File: test/board_test.rb [hunk 2]
```diff
@@ -14,31 +14,43 @@ def setup
...
   end
 
+  def test_ship_does_not_overlap
+    @board.place(@cruiser, ["A1", "A2", "A3"])
+
+    assert_equal false, @board.place(@submarine, ["A1", "B1"])
+    assert_equal @board.cells["A1"].ship, @cruiser
+    assert_nil @board.cells["B1"].ship
+  end
+
```
Comment (lrs8810) [inline test/board_test.rb]: These look awesome! You don't need to change this but @board.cells["A1".ship is the actual and the @cruiser is the expected. For assert_equal I believe it is expected, actual. We can keep this and refactor later for consistency if we want to.

File: test/board_test.rb [hunk 3]
```diff
@@ -76,4 +88,20 @@ def test_coordinate_letters_are_consecutive
...
   end
+
+  def test_the_ship_coordinate_is_not_diagonal
+    assert_equal false, @board.valid_placement?(@cruiser, ["A1", "B2", "C3"])
+    assert_equal false, @board.valid_placement?(@submarine, ["C2", "D3"])
+    assert_equal true, @board.valid_placement?(@cruiser, ["A1", "B1", "C1"])
+  end
+
+  def test_ship_can_be_placed
+    @board.place(@cruiser, ["A1", "A2", "A3"])
```
Comment (lrs8810) [inline test/board_test.rb]: This looks awesome! Great work!

File: test/board_test.rb (hunk 2 above)
Comment (sasloan) [reply test/board_test.rb]: Oh!! you're right!!! I didn't even notice that. should I refactor now? or can we just merge the branch and update it in our next merge?

**Pull request merged**